unittests/test_generator.py

```diff
@@ -0,0 +1,46 @@
+"""
+Unit test class for the Generator module.
+"""
+
+import os
+import platform
+import sys
+import unittest
+from pathlib import Path
+
+sys.path.insert(0, str(Path(__file__).resolve().parents[1] / "src"))
+
+from bo4egenerator import generator  # pylint: disable=wrong-import-position
+
+
+class TestGenerator(unittest.TestCase):
+    """
+    Unit test class for the Generator module.
+    """
+
+    def test_generate_csharp_classes(self) -> None:
+        """
+        Test case for generating C# classes using the `generate_csharp_classes` method.
+
+        This test case verifies that the `generate_csharp_classes` method correctly generates C# classes.
+        The Angebot schema file is used as an example for the test.
+        It also checks if the generated `Angebot.cs` file exists in the output directory.
+        """
+        project_root = Path.cwd() / "unittests" / "test-data"
```

Comment:
das cwd sorgt für unterschiedliches verhalten von unittest und coverage. Macht doch statdessen:
```suggestion
        project_root = Path(__file__).parent / "test-data"
```
dann bist du unabhängig vom cwd

Reply:
außerdem ist es nicht der project root ;) sondern der test-data root.

Reply:
sehr schön ^^